src/NHibernate/Async/Context/ThreadLocalSessionContext.cs

```diff
@@ -32,18 +32,19 @@ private static async Task CleanupAnyOrphanedSessionAsync(ISessionFactory factory
 
 				try
 				{
-					if (orphan.Transaction != null && orphan.Transaction.IsActive)
+					try
 					{
-						try
+						var rollbackTask = orphan.CurrentTransaction?.RollbackAsync(cancellationToken);
```

Comment:
Are you sure `IsActive` check is no more needed here?

Reply:
It was previously needed because the `Transaction` was anyway never null, and was an inactive one if no transaction was ongoing.
Due to the `Transaction` property being not yet removed, it can still be created without being started altogether, so in fact the `IsActive` check should remain for now.

Reply:
> It was previously needed because the Transaction was anyway never null, and was an inactive one if no transaction was ongoing.

`ITransaction.IsActive` might contain some custom logic for determination. That's my main concern here.. To me it seems `IsActive` check should remain even after  `Transaction` removal.

Reply:
It seems highly unlikely to me that such an implementation would be supported by NHibernate. It is not expected that a transaction might go inactive without an explicit commit or rollback call and without still supporting such a call if it goes inactive "by itslef". Most code just rollback or commit transactions it has explicitly started without checking that flag.

Reply:
But anyway, we can leave that for further reconsideration when `Transaction` properties would be actually removed.